src/core/qgscolorramp.cpp

```diff
@@ -401,7 +402,7 @@ QList<QColor> QgsLimitedRandomColorRamp::randomColors( int count,
   int safeValMin = std::min( valMin, valMax );
 
   //start hue at random angle
-  double currentHueAngle = 360.0 * static_cast< double >( qrand() ) / RAND_MAX;
+  double currentHueAngle = 360.0 * static_cast< double >( QRandomGenerator::system()->generate() ) / RAND_MAX;
```

Comment:
We should be using "global" not "system" in these methods. System is for cryptography,. global is for bulk number generation.